ArchiSteamFarm/www/pages/commands.html

```diff
@@ -321,59 +322,105 @@ <h3 class="box-title" data-i18n="global-commands">Commands</h3>
-                        $('#iconBotsDropDownBox').removeClass('fa-caret-down');
-                        $('#iconBotsDropDownBox').addClass('fa-caret-up');
+						$commandsDropDown.addClass('dropup');
+						$botsDropDown.addClass('dropup');
+						$commandsDropDown.selectpicker('refresh');
+						$botsDropDown.selectpicker('refresh');
                     } else {
-                        $cmdsDropDownBox.removeClass('dropup');
-                        $botsDropDownBox.removeClass('dropup');
-                        $('#iconCmdsDropDownBox').removeClass('fa-caret-up');
-                        $('#iconCmdsDropDownBox').addClass('fa-caret-down');
-                        $('#iconBotsDropDownBox').removeClass('fa-caret-up');
-                        $('#iconBotsDropDownBox').addClass('fa-caret-down');
+						$commandsDropDown.removeClass('dropup');
+						$botsDropDown.removeClass('dropup');
+						$('.bootstrap-select').removeClass('dropup');
+						$commandsDropDown.selectpicker('refresh');
+						$botsDropDown.selectpicker('refresh');
                     }
                 });
 
             $('.box-footer').ready(function () {
-                for (var cmd in myCommands) { // Fill drop down with all commands
-                    if (myCommands.hasOwnProperty(cmd)) {
-                        $commandsDropDown.append('<li><a href="javascript:void(0)" id="' + cmd + '" onclick="fillCommand(\'' + cmd + '\')">' + cmd + '</a></li>');
-
-                        $('#' + cmd)
-                            .mouseenter(function () {
-                                $commandInput.attr('placeholder', myCommands[this.id]); // ToDo: Find a better solution for where to display description
-                            })
-                            .mouseleave(function () {
-                                $commandInput.attr('placeholder', $.i18n('placeholder-commands'));
-                            });
-                    }
-                }
-
-                $botsDropDown.append('<li><a href="javascript:void(0)" onclick="fillBots(\'ASF\')">ASF</a></li>'); // Fill with default value
-
-                $.ajax({ // Fill drop down with all bots + ASF
+				var cachedCommands = get('cachedCommands'),
+					cachedDate = get('cachedCommandsDate'),
+					today = new Date().toLocaleDateString();
+				
+				if (cachedCommands && today == cachedDate) {
+					var myCommands = cachedCommands.split(',');
+				
+					for (var i = 0; i < myCommands.length; i++) {
+						$commandsDropDown.append('<option title="Commands">' + myCommands[i] + '</option>');
+					}
+					
+					$('.overlay').remove();
+					$commandsDropDown.selectpicker('refresh');
+				} else {
+					$('#commandsDropDownBox').append('<div class="overlay bot-box-loading"><i class="fas fa-sync fa-spin"></i></div>');
+				
+					$.ajax({
+						url: '/Api/WWW/Send',
+						type: 'POST',
+						data: JSON.stringify({ "URL": "https://github.com/JustArchi/ArchiSteamFarm/wiki/Commands" }),
```

Comment:
I mean both approaches are correct and will work, but by doing it raw you skip all html related to GitHub and all things that need to be loaded and rendered. This is your choice, but it'd be great if you checked out if raw way isn't more effective.

Reply:
It would be faster but right now I dont know how to parse githubs markdown.

I think for now we can leave it like this and in the future I will improve it.

This whole function isnt the cleanest, so I assume it will change in future anyway.

Reply:
I found something like that - https://github.com/showdownjs/showdown / http://demo.showdownjs.com/

Reply:
The least you can do is probably to just fetch raw, put it in markdown -> html converter and extract via xpath, this doesn't look too problematic compared to what you have right now, but once again your call.

Reply:
I will definitely look into it but thats something for a future update. :+1:

Reply:
Alright, thanks for everything like usual 👍